example/dynamic.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/inconshreveable/log15"
+	"github.com/ngrok/libngrok-go"
+)
+
+func unwrap[T any](out T, err error) T {
+	if err != nil {
+		log15.Error("unwrapped error", "error", err)
+		os.Exit(1)
+	}
+
+	return out
+}
+
+func httpResp(w http.ResponseWriter, code int, msg string, args ...any) {
+	w.WriteHeader(code)
+	fmt.Fprintf(w, msg, args...)
+}
+
+func main() {
+	log15.Root().SetHandler(log15.LvlFilterHandler(log15.LvlInfo, log15.StdoutHandler))
+
+	restart := true
+	domain := ""
+
+	ctx := context.Background()
+
+	for restart {
+		sess := unwrap(libngrok.Connect(ctx, libngrok.ConnectOptions().
+			WithLog15(log15.Root()).
+			WithRemoteCallbacks(libngrok.RemoteCallbacks{
+				OnStop: func(ctx context.Context, sess libngrok.Session) error {
+					restart = false
+					log15.Info("exiting due to remote request")
+					return nil
+				},
+				OnRestart: func(ctx context.Context, sess libngrok.Session) error {
+					log15.Info("restarting due to remote request")
+					return nil
+				},
+			}).
+			WithAuthToken(os.Getenv("NGROK_TOKEN")),
+		))
+
+		tun := unwrap(sess.StartTunnel(ctx, libngrok.HTTPOptions().
+			WithDomain(domain).
```

Comment:
this is an empty string when it first is ran, is this just so if it restarts on accident you re-use the same random domain name you got the first time?

Reply:
Yeah, this only makes sense with the reconnect cookie, which we're removing from the public API.

So the domain arg here will go away as well.